source/postprocess/visualization/volumetric_strain_rate.cc

```diff
@@ -0,0 +1,89 @@
+/*
+  Copyright (C) 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/visualization/volumetric_strain_rate.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      VolumetricStrainRate<dim>::
+      VolumetricStrainRate ()
+        :
+        DataPostprocessorScalar<dim> ("volumetric_strain_rate",
+                                      update_gradients | update_q_points)
+      {}
+
+
+      template <int dim>
+      void
+      VolumetricStrainRate<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double> > &computed_quantities) const
+      {
+        const unsigned int n_quadrature_points = input_data.solution_values.size();
+        Assert (computed_quantities.size() == n_quadrature_points,    ExcInternalError());
+        Assert (computed_quantities[0].size() == 1,                   ExcInternalError());
+        Assert (input_data.solution_values[0].size() == this->introspection().n_components,           ExcInternalError());
+        Assert (input_data.solution_gradients[0].size() == this->introspection().n_components,          ExcInternalError());
+
+        for (unsigned int q=0; q<n_quadrature_points; ++q)
+          {
+            // extract the velocity gradients
+            double div_u = 0.0;
+            Tensor<2,dim> grad_u;
+            for (unsigned int d=0; d<dim; ++d)
+              {
+                grad_u[d] = input_data.solution_gradients[q][d];
+                div_u += grad_u[d][d];
```

Comment:
I can do that, I just thought if I write it like this, the code is easier to understand.